src/stitch.rs

```diff
@@ -85,25 +86,39 @@ impl Iterator for IterStitchedIndexHunks {
                         return Some(hunk);
```

Comment:
This debug assert will fail on the first iteration.  
`band_id` will be set (except if there are no previous bands in the archive) but `index_hunks` will be `None` since the band hasn't been opened and iterated yet.  
I see what you're trying to ensure. Sense full will be an assert before the `index_hunks` for-loop, ensuring that if `index_hunks` is `Some`, `band_id` (indicating the band of the current index) is `Some` as well.